editor/editor_file_system.cpp

```diff
@@ -1748,7 +1749,8 @@ String EditorFileSystem::_get_global_script_class(const String &p_type, const St
 void EditorFileSystem::_update_file_icon_path(EditorFileSystemDirectory::FileInfo *file_info) {
 	String icon_path;
 	if (file_info->script_class_icon_path.is_empty() && !file_info->deps.is_empty()) {
-		const String &script_path = file_info->deps[0]; // Assuming the first dependency is a script.
+		const String &script_dep = file_info->deps[0]; // Assuming the first dependency is a script.
+		const String &script_path = script_dep.contains("::") ? script_dep.get_slice("::", 2) : script_dep;
```

Comment:
Are you sure 2 is the correct slice here? Whenever I encounter subresources, they are of the form `res://some/path::Something_asdf` - I don't think I've ever seen two sets of `::`

My guess is it should be slice 0, just like the other example above.

Would it be able to add a comment with an example of such a `res://` path so I can better understand what is being parsed here?

Reply:
This is not about subresources. Like above, `deps` contains the output of `ResourceLoader*.get_dependencies()`, which is in one of the following formats:
1. `uid::type::path`
2. `path::type` if no uid exists and type not requested,
3. just path if no uid and path not requested.

It could fail in the second case, but when being set type is never requested, so currently case 2 is not happening

Reply:
As for why, I decided to go with path instead of uid, it's because it's used for efsd `find_file`, which can't use uid

Reply:
https://github.com/godotengine/godot/blob/4ab8fb809396fa38ba929fec97cfcb7193f1c44d/scene/resources/resource_format_text.cpp#L886-L899

Reply:
Pretty sure the format is always `uid::type::path`. If any is omitted, one part will be empty, e.g. `uid::::path`.